src/services/pool.ts

```diff
@@ -0,0 +1,112 @@
+import { Db, Filter, FindOptions, MongoClient, WithId } from 'mongodb';
+import { createMessage, initiateLogger } from './logger';
+import { assertTrailingSlash } from '../utils';
+
+/** BEGIN typing for DB */
+
+type EnvKeyedObject = {
+  prd: string;
+  preprd: string;
+  dotcomstg: string;
+  dotcomprd: string;
+};
+
+interface BranchEntry {
+  name: string;
+  gitBranchName: string;
+  active: boolean;
+  urlAliases: string[];
+  [key: string]: any;
+}
+
+interface RepoDocument extends WithId<Document> {
+  repoName: string;
+  project: string;
+  branches: BranchEntry[];
+  url: EnvKeyedObject;
+  prefix: EnvKeyedObject;
+  search: Record<string, string>;
+}
+
+interface BranchResponse {
+  gitBranchName: string;
+  active: boolean;
+  fullUrl: string;
+}
+
+interface RepoResponse {
+  repoName: string;
+  project: string;
+  search: Record<string, string>;
+  branches: BranchResponse[];
+}
+
+/** END typing for DB */
+
+const ATLAS_URI = process.env.ATLAS_URI ?? '';
+const REPOS_COLLECTION = 'repos_branches';
+const DB_NAME = process.env.POOL_DB_NAME ?? 'pool';
+const ENV_URL_KEY = (process.env.SNOOTY_ENV ?? 'dotcomprd') as keyof EnvKeyedObject;
+
+const logger = initiateLogger();
+const client = new MongoClient(ATLAS_URI);
+let dbInstance: Db | null = null;
+
+// exposes db instance
+export const poolDb = async () => {
+  if (dbInstance) return dbInstance;
+
+  try {
+    await client.connect();
+    dbInstance = client.db(DB_NAME);
+    return dbInstance;
+  } catch (e) {
+    logger.error(createMessage(`Error while connecting client: ${e}`));
+    throw e;
+  }
+};
+
+export const closeDBConnection = async () => {
+  await client.close();
+};
+
+export const findAllRepos = async (query: Filter<RepoDocument> = {}, options: FindOptions = {}, reqId?: string) => {
```

Comment:
(No change needed) Do we need a query param if this function is always intended to return all repo data? Just wondering if we ever see this function being used for custom finds

Reply:
good call. since its `findAll` will remove the query param